packages/core/src/utilities/getCurrentVolumeViewportSlice.ts

```diff
@@ -0,0 +1,168 @@
+import { glMatrix, mat4, vec3 } from 'gl-matrix';
+import { transformIndexToWorld, transformWorldToIndex } from '.';
+import { IVolumeViewport, Point2, Point3 } from '../types';
+
+/**
+ * Get the image data for the current slice rendered on the viewport.
+ * It does not work for oblique views.
+ * @param viewport - Volume viewport
+ * @returns Slice image dataand  matrices to convert from volume
+ *   to slice and vice-versa
+ */
+function getCurrentVolumeViewportSlice(viewport: IVolumeViewport) {
```

Comment:
I feel like we can generalize it to be `getCurrentViewImageData(viewport)`, since for stack viewport we can grab that from .getImageData right?

Also we should throw error for oblique

Reply:
That method does not have any stack-related code.

Reply:
It throws an error a few lines below